capz/run-capz-e2e.sh

```diff
@@ -181,6 +182,254 @@ run_e2e_test() {
             --v=5 "${ADDITIONAL_E2E_ARGS[@]}"
         set +x
         log "e2e tests complete"
+
+        if [[ "$INCLUDE_NPM_TESTS" == "true" ]]; then
+            # write all results to npm-tests directory
+            mkdir -p npm-tests
+            cd npm-tests
+            run_all_npm_tests
+            cd ..
+        fi
+    fi
+}
+
+
+run_all_npm_tests () {
```

Comment:
can we put all the npm stuff in a separate script / file?
This is adding a lot of bulk to this main script that is not related to primary function